src/intervals/interval.jl

```diff
@@ -0,0 +1,175 @@
+
+# Base interval types and utilities
+# ---------------------------------
+bitstype 8 Strand
+
+convert(::Type{Strand}, strand::Uint8) = box(Strand, unbox(Uint8, strand))
+convert(::Type{Uint8}, strand::Strand) = box(Uint8, unbox(Strand, strand))
+
+const STRAND_NA   = convert(Strand, 0b000)
+const STRAND_POS  = convert(Strand, 0b001)
+const STRAND_NEG  = convert(Strand, 0b010)
+const STRAND_BOTH = convert(Strand, 0b011)
+
+function show(io::IO, strand::Strand)
+    if strand == STRAND_NA
+        print(io, "(indeterminate strand)")
+    elseif strand == STRAND_POS
+        print(io, "+")
+    elseif strand == STRAND_NEG
+        print(io, "-")
+    elseif strand == STRAND_BOTH
+        print(io, ".")
+    else
+        print(io, "(undefined strand)")
+    end
+end
+
+
+function isless(a::Strand, b::Strand)
+    return convert(Uint8, a) < convert(Uint8, b)
+end
+
+
+# Note, just to be clear: this shadows IntervalTrees.Interval
+@doc """
+A genomic interval specifies interval with some associated metadata.
+""" ->
+immutable Interval{T} <: IntervalTrees.AbstractInterval{Int64}
```

Comment:
Since `IntervalTrees` export `AbstractInterval`, we don't need to write explicitly the module name here, do we?